deployments/hadoop-yarn/ansible/31-create-monitor.yml

```diff
@@ -0,0 +1,96 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2020, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#
+
+- name: "Create our Monitor node"
+  gather_facts: false
+  hosts: localhost
+  vars_files:
+    - /tmp/ansible-vars.yml
+  tasks:
+
+    - name: "Create a security group for our monitor"
+      os_security_group:
+        cloud: "{{ cloudname }}"
+        state: present
+        name: "{{ security['monitor'] }}"
+      register:
+        monitorsec
+
+    - name: "Allow 9090 from any"
+      os_security_group_rule:
+        cloud: "{{ cloudname }}"
+        state: present
+        security_group: "{{ security['monitor'] }}"
+        direction: 'ingress'
+        protocol:  'tcp'
+        port_range_min: 9090
+        port_range_max: 9090
+
+    - name: "Allow 9093 from any"
+      os_security_group_rule:
+        cloud: "{{ cloudname }}"
+        state: present
+        security_group: "{{ security['monitor'] }}"
+        direction: 'ingress'
+        protocol:  'tcp'
+        port_range_min: 9093
+        port_range_max: 9093
+
+
+    - name: "Allow 3000 from any"
+      os_security_group_rule:
+        cloud: "{{ cloudname }}"
+        state: present
+        security_group: "{{ security['monitor'] }}"
+        direction: 'ingress'
+        protocol:  'tcp'
+        port_range_min: 3000
+        port_range_max: 3000
```

Comment:
Can we make this more specific ? Do we know which nodes need to connect on these ports ?
I thought most of the Prometheus monitoring was pulled (scraped) by the monitor node rather than pushed by the target nodes.

Reply:
The following commit specifies what those ports are opened for: https://github.com/wfau/aglais/pull/457/commits/45a47d09b4a24601b72ecc59fb11509e9d8695ee#diff-4329c2dd33ce85e64f39644bd3596958015f6e47fdadcb259e9123dd31b8484d
(Basically the Prometheus / Grafana / AlertManager UIs)

Reply:
If you are not assigning a public IP address to the monitoring node, then which machines need to access these ports ?

Reply:
If you are planning on using a `ssh` tunnel via the Zeppelin node, then all external connections will be coming from there. So rather than `all` how about limit acces to just the `zeppelin` node and add a comment in the code to explain why ?

Reply:
New commit to only allow access from zeppelin: https://github.com/wfau/aglais/pull/457/commits/37d8b472c87e109a84e49b761f5d55713c124ba1